design-documents/asynchronous-operations/poison-pill.md

```diff
@@ -0,0 +1,20 @@
+## Glossary
+
+poison pill
+: message in Data Storage that signals that consumer should die
+
+### Overview
+
+Right now our consumer depends on config/objects in-memory state. After application was bootstrapped it initialize a lot of object that are always kept in memory after first load - config, websites, etc.., since our consumers are long-living and their live time limited only by message count it leads to wrong im-memory config/objects states and we need to introduce the way to reinit them.
```

Comment:
Could you clarify what "wrong" means? Is it a state modified by the processes, so it is changed?

Reply:
wrongs means that if someone will change the state, consumer will running with outdated in cache, yeah outdated is more accurate meaning right here